DRAFT.md

```diff
@@ -1173,25 +1173,39 @@ outside of tests.
 
 A converting constructor could be added in a future version of the C++ standard.
 
-### Small object optimization for `polymorphic`
-
-`polymorphic` could be designed to include a small object optimization like that
-used in `std::string`. A small object optimization uses a buffer to potentially
-store the owned object and avoid allocating memory. This would make move
-construction more complicated as the owned object must be moved from one buffer
-to another, potentially invoking allocations if the owned object's move
-constructor allocates memory.
-
-As designed, `polymorphic<T>` does not require that `T` (or constructed classes
-of type `U` derived from `T`) are move constructible or move assignable for
-`polymorphic<T>` to be move constructible or move assignable.
-
-A polymorphic value type with a small buffer optimization that did not allocate
-a control block for the owned object would need to be a different type. It is
-not possible to add a small object optimization to `polymorphic` without making
-breaking changes. There may be a case for the addition of `small_polymorphic<T,
-N>` similar to `llvm::SmallVector<T, N>`, but we are not proposing its addition
-here.
+### Small Buffer Optimisation
+
+It is possible to implement `polymorphic` with a small buffer optimisation,
+similar to that used in `std::function`. This would allow `polymorphic` to store
+small objects without allocating memory. Like `std::function`, the size of the
+small buffer is left to be specified by the implementation.
+
+The authors are sceptical of the value of a small buffer optimisation for
+objects from a type hierarchy. If the buffer is too small, all instances of
+`polymorphic` will be larger than needed. This is because they will allocate
+heap in addition to having the memory from the (empty) buffer as part of the
+object size. If the buffer is too big, `polymorphic` objects will be larger than
+necessary, potentially introducing the need for `indirect<polymorphic<T>>`.
+
+We could add another, non-type, template argument to `polymorphic` to specify
+the size of the small buffer:
+
+```c++
+template <typename T, typename Alloc, size_t BufferSize>
+class polymorphic;
+```
+
+We opt not to provide an additional template argument to maintain consistency
```

Comment:
Minor rewording: 

However we opt not to do this to maintain consistency with other standard library types.